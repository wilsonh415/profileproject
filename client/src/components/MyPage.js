import React from 'react';
import NavbarComp from './NavbarComp';
import { BrowserRouter, Route } from 'react-router-dom';
import Resume from './Resume';
import Projects from './Projects';
import Random from './Random';
import Home from './Home';
import Iss from './projectComponents/Iss';
import MinesweeperGame from './projectComponents/Game';
 
class MyPage extends React.Component {
   render() {
       return (
           <div className="thePage">
               <div className="bgBlock">
               </div>
               <NavbarComp></NavbarComp>
               <BrowserRouter>
                   <Route component={Home} exact path="/"></Route>
                   <Route component={Resume} path="/Resume"></Route>
                   <Route component={Projects} path="/Projects"></Route>
                   <Route component={Random} path="/Random"></Route>
                   <Route component={Iss} path="/Iss"></Route>
                   <Route component={MinesweeperGame} path="/Minesweeper"></Route>
               </BrowserRouter>
           </div>
       )
   }
}
 
export default MyPage;
