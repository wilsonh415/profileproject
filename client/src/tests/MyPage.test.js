import React from 'react';
import { shallow } from 'enzyme';
import { configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import MyPage from '../components/MyPage';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test('MyPage component renders correctly', () => {
    const component = renderer.create(<MyPage></MyPage>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});