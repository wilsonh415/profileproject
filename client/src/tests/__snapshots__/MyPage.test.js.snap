// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPage component rendered properly 1`] = `
<div
  className="thePage"
>
  <div
    className="bgBlock"
  />
  <div
    className="navigationBar"
  >
    <nav
      className="navbar navbar-expand navbar-light bg-lightgrey"
    >
      <div
        className="border-left mx-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/Resume"
          href="/Resume"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <b
            style={
              Object {
                "backgroundColor": "skyblue",
                "color": "black",
                "fontFamily": "Palatino Linotype",
                "fontSize": "21px",
                "marginRight": "2vw",
                "opacity": 0.9,
              }
            }
          >
            Resume
          </b>
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/Projects"
          href="/Projects"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <b
            style={
              Object {
                "backgroundColor": "skyblue",
                "color": "black",
                "fontFamily": "Palatino Linotype",
                "fontSize": "21px",
                "marginRight": "2vw",
                "opacity": 0.9,
              }
            }
          >
            Projects
          </b>
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/Random"
          href="/Random"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <b
            style={
              Object {
                "backgroundColor": "skyblue",
                "color": "black",
                "fontFamily": "Palatino Linotype",
                "fontSize": "21px",
                "marginRight": "2vw",
                "opacity": 0.9,
              }
            }
          >
            Random
          </b>
        </a>
      </div>
    </nav>
  </div>
  <div
    className="homeStyle"
  >
    <h1
      style={
        Object {
          "color": "white",
          "fontFamily": "Georgia",
          "fontSize": "45px",
          "marginLeft": "10vw",
          "marginRight": "10vw",
          "position": "relative",
          "top": "19%",
        }
      }
    >
      Hi, I'm Wilson. Welcome to my website!
    </h1>
  </div>
</div>
`;
